src/main/java/traffic/TrafficLightColour.java

```diff
@@ -0,0 +1,15 @@
+package main.java.traffic;
+
+//An enumeration of the colors of a traffic light. 
+enum TrafficLightColour {
+	RED(1000), RED_YELLOW(500), GREEN(1000), YELLOW(1000);
```

Comment:
Seems like an awkward way of getting around the book version's lack of amber after green. Given that these are public fields, it could be confusing to somone using it. You could get around it using a private static array of state transitions and a method providing an iterator over them?

```java
private static final TrafficLightColour[] STATE_TRANSITIONS = { RED, YELLOW, GREEN, YELLOW };

public static Iterator<TrafficLightColour> iterator() {
  return new Iterator<TrafficLightColour>() {
    int i = 0;
    boolean hasNext() { return true; }
    TrafficLightColour next() { return STATE_TRANSITIONS[i = ++i % STATE_TRANSITIONS.length]; }
  };
}
```

I haven't tested this though, it might complain about using constants before they've been loaded.